contracts/test/BoundlessMarket.t.sol

```diff
@@ -828,20 +829,43 @@ contract BoundlessMarketBasicTest is BoundlessMarketTest {
         return _testLockRequestBadClientSignature(false);
     }
 
-    function testLockRequestWithSignatureBadProverSignature() public {
+    function testLockRequestWithSignatureProverSignatureIncorrectRequest() public {
         Client client = getClient(1);
         ProofRequest memory request = client.request(1);
         bytes memory clientSignature = client.sign(request);
-        // Bad signature is over the wrong request.
-        bytes memory badProverSignature = testProver.sign(client.request(2));
+        // Prover signs the incorrect request.
+        bytes memory badProverSignature = testProver.signLockRequest(LockRequest({request: client.request(2)}));
 
         // NOTE: Error is "InsufficientBalance" because we will recover _some_ address.
         // It should be random and never correspond to a real account.
         // TODO: This address will need to change anytime we change the ProofRequest struct or
         // the way it is hashed for signatures. Find a good way to avoid this.
         vm.expectRevert(
             abi.encodeWithSelector(
-                IBoundlessMarket.InsufficientBalance.selector, address(0x6519523979d391bF62ba1dCcEFf81BF484eB067b)
+                IBoundlessMarket.InsufficientBalance.selector, address(0x363482aC3961d6dEdFC95d7f496a5d2271A6c19f)
+            )
+        );
+        boundlessMarket.lockRequestWithSignature(request, clientSignature, badProverSignature);
+
+        client.expectBalanceChange(0 ether);
+        testProver.expectBalanceChange(0 ether);
+        expectMarketBalanceUnchanged();
+    }
+
+    function testLockRequestWithSignatureProverSignatureIncorrectDomain() public {
+        Client client = getClient(1);
+        ProofRequest memory request = client.request(1);
+        bytes memory clientSignature = client.sign(request);
+        // Prover signs ProofRequest struct rather than LockRequest struct.
```

Comment:
```suggestion
        // Prover signs ProofRequest struct rather than LockRequest struct.
        // NOTE: This was how the contract worked in a previous version. This is included as a regression test.
```